parachain-tracer/src/prometheus.rs

```diff
@@ -207,6 +212,28 @@ impl Metrics {
 		}
 	}
 
+	/// Update on-demand latency in blocks
+	pub(crate) fn on_on_demand_delay(&self, delay_blocks: u32, para_id: u32) {
```

Comment:
These functions could use better naming:

```suggestion
	pub(crate) fn handle_on_demand_delay(&self, delay_blocks: u32, para_id: u32) {
```